scripts/download_hf_assets.py

```diff
@@ -0,0 +1,235 @@
+                        matches_found = True
+                        break
+
+                if not matches_found:
+                    print(
+                        f"Warning: No matching files found for asset_type '{asset_type}' in {repo_id}"
+                    )
+
+        if not files_found:
+            print(f"Warning: No matching files found in {repo_id}")
+            print(f"Available files: {available_files[:10]}...")
+            return
+
+        print(f"Found {len(files_found)} files:")
+        for f in files_found:
+            print(f"  - {f}")
+
+        downloaded_files = []
+        missed_files = []
+
+        # Download files with progress bar
+        with tqdm(
+            total=len(files_found), desc="Downloading files", unit="file"
+        ) as pbar:
+            for filename in files_found:
+                try:
+                    pbar.set_description(f"Downloading {os.path.basename(filename)}")
+
+                    hf_hub_download(
+                        repo_id=repo_id,
+                        filename=filename,
+                        local_dir=model_dir,
+                        token=hf_token,
+                    )
+                    downloaded_files.append(filename)
+                    pbar.update(1)
+
+                except HTTPError as e:
+                    if e.response and e.response.status_code == 404:
+                        print(f"File {filename} not found, skipping...")
+                        missed_files.append(filename)
+                        pbar.update(1)
+                        continue
+                    else:
+                        raise e
+
+        if downloaded_files:
+            print(
+                f"\nSuccessfully downloaded {len(downloaded_files)} files to: {model_dir}"
+            )
+        if missed_files:
+            print(f"Warning: Some files could not be downloaded: \n{missed_files}")
+
+    except HTTPError as e:
+        if e.response and e.response.status_code == 401:
+            print(
+                "You need to pass a valid `--hf_token=...` to download private checkpoints."
+            )
+        raise e
+
+
+if __name__ == "__main__":
+    import argparse
+
+    parser = argparse.ArgumentParser(
+        description="Download files from HuggingFace Hub. "
+        "Automatically detects and downloads common tokenizer files (tokenizer.json, "
+        "tokenizer_config.json, tokenizer.model, ...) that work with Tokenizer."
+    )
+    parser.add_argument(
+        "--repo_id",
+        type=str,
+        required=True,
+        help="Repository ID to download from (e.g., 'meta-llama/Llama-3.1-8B', 'deepseek-ai/DeepSeek-V3')",
+    )
+    parser.add_argument(
+        "--hf_token",
+        type=str,
+        default=None,
+        help="HuggingFace API token (required for private repos)",
+    )
+    parser.add_argument(
+        "--local_dir",
+        type=str,
+        default="assets/hf/",
```

Comment:
this is going to change the directory for the tokenizers so then all of the `.toml` config files will also have to change?